test/k8sT/Istio.go

```diff
@@ -0,0 +1,171 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8sTest
+
+import (
+	"fmt"
+
+	. "github.com/cilium/cilium/test/ginkgo-ext"
+	"github.com/cilium/cilium/test/helpers"
+
+	. "github.com/onsi/gomega"
+	"github.com/sirupsen/logrus"
+)
+
+var _ = Describe("K8sValidatedIstioTest", func() {
+
+	var (
+		kubectl          *helpers.Kubectl
+		logger           *logrus.Entry
+		microscopeErr    error
+		microscopeCancel func() error
+		ciliumPodK8s1    string
+		ciliumFilter     string = "-l k8s-app=cilium"
+		ciliumPath       string = helpers.ManifestGet("cilium_ds_istio.yaml")
+		istioManifest    string = helpers.ManifestGet("istio-auth.yaml")
+		istioNS          string = "istio-system"
+		istioFilter      string = "-l zgroup=istio"
+		istioVersion     string = "release-0.8-20180521-15-16"
+		istioURL         string = fmt.Sprintf("https://storage.googleapis.com/istio-prerelease/daily-build/%[1]s/istio-%[1]s-linux.tar.gz", istioVersion)
+		istioConfigMap   string = helpers.ManifestGet("istio-kube-inject-template.yaml")
+		istioInitPolicy  string = helpers.ManifestGet("istio-sidecar-init-policy.yaml")
+		demoPath         string = helpers.ManifestGet("demo_big.yaml")
+		container        string = "app-frontend"
+	)
+
+	BeforeAll(func() {
+		logger = log.WithFields(logrus.Fields{"testName": "K8sValidatedIstioTest"})
+		logger.Info("Starting")
+
+		kubectl = helpers.CreateKubectl(helpers.K8s1VMName(), logger)
+		kubectl.Apply(ciliumPath)
+
+		_ = kubectl.Exec(fmt.Sprintf(
+			"%s -n %s delete pods %s",
+			helpers.KubectlCmd, helpers.KubeSystemNamespace, ciliumFilter))
+
+		err := kubectl.WaitforPods(helpers.KubeSystemNamespace, ciliumFilter, 600)
+		Expect(err).Should(BeNil())
+
+		err = kubectl.WaitKubeDNS()
+		Expect(err).Should(BeNil(), "DNS is not ready after timeout")
+
+		_, err = kubectl.CiliumPolicyAction(
+			helpers.KubeSystemNamespace, istioInitPolicy,
+			helpers.KubectlApply, 300)
+		Expect(err).Should(BeNil(), "Cannot apply init policy")
+
+		ciliumPodK8s1, err = kubectl.GetCiliumPodOnNode(helpers.KubeSystemNamespace, helpers.K8s1)
+		Expect(err).Should(BeNil(), "Cannot get cilium pod on k8s1")
+
+		// Installing Istio
+		res := kubectl.Exec(fmt.Sprintf("cd /tmp/; curl -L %s | tar xz ", istioURL))
+		res.ExpectSuccess("Cannot download istio: %s", res.CombineOutput())
+
+		res = kubectl.ExecWithSudo(fmt.Sprintf("mv -f /tmp/istio-%s /usr/local/istio", istioVersion))
+		res.ExpectSuccess("Cannot mv Istio source folder ")
+
+		res = kubectl.ExecWithSudo("ln -sf /usr/local/istio/bin/* /usr/local/bin/")
+		res.ExpectSuccess("cannot copy Istio binaries")
+
+		kubectl.Apply(istioManifest)
+
+		err = kubectl.WaitforPods(istioNS, istioFilter, 600)
+		Expect(err).Should(BeNil(), "Istio cannot be installed correctly")
+	})
+
+	kubeInject := func(fileName string, command string) *helpers.CmdRes {
```

Comment:
Sync up with @rlenglet about how to install Istio; it may be much simpler since we use Helm in the Istio GSG now.